### PR TITLE
Implement Forgot Password & Reset Password workflows

### PR DESCRIPTION
**Pull request by JordanGuinn**
Closes https://github.com/ctoec/data-collection/issues/840.

### CODE REVIEW

File: src/WingedKeys/Quickstart/Account/AccountController.cs [hunk 1]
```diff
@@ -207,6 +210,67 @@ public IActionResult AccessDenied()
...
+            {
+                var user = await _userManager.FindByEmailAsync(model.Email);
+
+                if (user != null && await _userManager.IsEmailConfirmedAsync(user))
+                {
+                    string token = await _userManager.GeneratePasswordResetTokenAsync(user);
+                    var callbackUrl = Url.Action(nameof(ResetPassword), "Account", new { email = user.Email, token = token }, protocol: Request.Scheme);
+
+                    await new EmailService().SendEmailAsync(model.Email, "Password Reset Request",
+                        "Please confirm your account by clicking <a href=\"" + callbackUrl + "\">here</a>.");
```
Comment (mmazanec22) [inline src/WingedKeys/Quickstart/Account/AccountController.cs:233]: Links that just say "here" are an accessibility issue-- the text should describe where the link goes or what it does.  So in this case something like
```
<a href="blablabla">Please confirm your account</a>
```

**Review by mmazanec22 — APPROVED** [review]

File: src/WingedKeys/Data/Migrations/WingedKeys/20201207234719_AddLoginDate.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace WingedKeys.Data.Migrations.WingedKeys
+{
+    public partial class AddLoginDate : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<DateTimeOffset>("LastLogin", "AspNetUsers", nullable: true);
```
Comment (jlhogan) [inline src/WingedKeys/Data/Migrations/WingedKeys/20201207234719_AddLoginDate.cs:10]: is the plan to use this to force a reset on first login?

Comment (JordanGuinn) [reply src/WingedKeys/Data/Migrations/WingedKeys/20201207234719_AddLoginDate.cs:10]: Yes!

**Review by jlhogan — APPROVED** [review]
wahooo :tada: Do we have a ticket for the force reset work?

File: src/WingedKeys/Views/Admin/TriggerInitialLogin.cshtml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      <div class="col-sm-6">
+        <strong>@(Model.Error)</strong>
+      </div>
+    </div>
+  }
+  <div class="row">
+    <div class="col-sm-6">
+      <div class="panel panel-default">
+        <div class="panel-heading">
+            <h3 class="panel-title">New account details</h3>
```
Comment (mmazanec22) [inline src/WingedKeys/Views/Admin/TriggerInitialLogin.cshtml:27]: This should be an h2 bc it's nested under an h1 (unless I'm missing something)

Comment (JordanGuinn) [reply src/WingedKeys/Views/Admin/TriggerInitialLogin.cshtml:27]: Nope - literally just copied and pasted this from the `Create new account` view

**Pull request merged**